src/Discord/Parts/Channel/Channel.php

```diff
@@ -552,37 +553,52 @@ public function deleteMessages($messages, ?string $reason = null): ExtendedPromi
             return resolve();
         } elseif ($count == 1 || $this->is_private) {
             foreach ($messages as $message) {
-                if ($message instanceof Message ||
-                    $message = $this->messages->get('id', $message)
-                ) {
+                if ($message instanceof Message) {
                     return $message->delete();
                 }
 
                 return $this->http->delete(Endpoint::bind(Endpoint::CHANNEL_MESSAGE, $this->id, $message));
             }
         } else {
             $messageID = [];
+            $oldMessageID = [];
+            $promises = [];
+
+            $headers = [];
+            if (isset($reason)) {
+                $headers['X-Audit-Log-Reason'] = $reason;
+            }
 
             foreach ($messages as $message) {
                 if ($message instanceof Message) {
-                    $messageID[] = $message->id;
+                    if ($message->timestamp->timestamp < time()-1209600) {
+                        $oldMessageID[] = $message->id;
+                    } else {
+                        $messageID[] = $message->id;
+                    }
+                } elseif (getSnowflakeTimestamp($message) < time()-1209600) {
+                    $oldMessageID[] = $message;
                 } else {
                     $messageID[] = $message;
                 }
             }
 
-            $promises = [];
-
-            $headers = [];
-            if (isset($reason)) {
-                $headers['X-Audit-Log-Reason'] = $reason;
+            if (count($messageID) == 1) {
+                // this is not older than 2 weeks, but need to delete it in individual endpoint
+                $oldMessageID[] = $messageID[0];
+                // empty the bulk
+                $messageID = [];
             }
 
             while (! empty($messageID)) {
                 $promises[] = $this->http->post(Endpoint::bind(Endpoint::CHANNEL_MESSAGES_BULK_DELETE, $this->id), ['messages' => array_slice($messageID, 0, 100)], $headers);
                 $messageID = array_slice($messageID, 100);
             }
```

Comment:
This old code actually bugs me, what if there are 101 messages, the last message would get
```
Discord\Http\Exceptions\RequestFailedException: Bad Request - {     "message": "Provided too few or too many messages to delete. Must provide at least 2 and at most 100 messages to delete.",     "code": 50016 }
```